http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/JsonPayloadIT.java

```diff
@@ -0,0 +1,184 @@
+package io.quarkus.ts.http.advanced.reactive;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Date;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+
+@QuarkusScenario
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class JsonPayloadIT {
+
+    @QuarkusApplication(classes = { FootballTeamResource.class, FootballTeam.class
+    }, properties = "oidcdisable.properties")
+    static RestService app = new RestService();
+
+    @Test
+    @Order(1)
+    public void sendSuccessfulJSONPayloadFromAFile() throws IOException {
+
+        String jsonPayload = FileUtils.readFileToString(
+                Paths.get("src", "test", "resources", "football-teams.json").toFile(),
+                StandardCharsets.UTF_8);
+
+        Response response = given()
+                .contentType(ContentType.JSON)
+                .body(jsonPayload)
+                .when()
+                .post("/resource/upload-football-json")
+                .then()
+                .statusCode(201)
+                .body("teams.size()", is(27))
+                .extract().response();
+        assertThat(response.body().asString(), containsString("Teams added successfully"));
+
+    }
+
+    @Test
+    public void sendLargeComplexJSONPayload() throws IOException {
+        String jsonPayload = FileUtils.readFileToString(
+                Paths.get("src", "test", "resources", "big_sample.json").toFile(),
+                StandardCharsets.UTF_8);
+
+        Response response = given()
+                .contentType(ContentType.JSON)
+                .body(jsonPayload)
+                .when()
+                .post("/resource/process-json")
+                .then()
+                .statusCode(jakarta.ws.rs.core.Response.Status.OK.getStatusCode())
+                .extract().response();
+        assertThat(response.body().asString(), containsString("Hello, Singleton Middleton! You have 4 unread messages."));
```

Comment:
Here I would not extracting response but use:
```
.statusCode(201)
.body(containsString("Hello, Singleton Middleton! You have 4 unread messages."));
```

Reply:
I still think that we don't need to extract the response and should do it somethink like this:
```
.then()
.statusCode(OK_STATUS_CODE)
.body(containsString("Hello, Singleton Middleton! You have 4 unread messages."));
```